### PR TITLE
make utils a subpackage of snmachine

### PR DESCRIPTION
**Pull request by heather999**
**IMPORTANT: Please create an issue first before opening a Pull Request.**
Linked to issue(s): #258 

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## What changes were proposed in this pull request?

Making the `utils` package a subpackage of `snmachine`.

How is the issue this PR is referenced against solved with this PR?

The `utils` directory has been moved under `snmachine` and corresponding code has been updated to reference `snmachine.utils` where appropriate.  

## How was this patch tested?
Locally, I modified my copies of the example notebooks to use `snmachine.utils` and ran through the first 3 notebooks. It is necessary for someone to go through all the notebooks and rerun them to make sure things are ok.

## Final Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I am up to date with `dev` branch of `snmachine` at the time of this PR

### CODE REVIEW

**Comment by Catarina-Alves** [comment]
@heather999 did you have the time to look at my comments regarding this PR?

File: snmachine/snclassifier.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 from sklearn import model_selection
 from sklearn.model_selection import PredefinedSplit, StratifiedKFold
 from sklearn.preprocessing import StandardScaler
-from utils import plasticc_utils
+from . utils import plasticc_utils
```
[the remark this answers is not in the document]
Comment (heather999) [reply snmachine/snclassifier.py]: I just updated this to `from snmachine.utils` and that works for me - but it would be great if you could test it too

**Comment by heather999** [comment]
@Catarina-Alves I think this is ready for testing by others.  I did the following:
Created a new installation of Miniconda with python 3.8
Cloned this repo and checked out this branch `general/issue/258/utils-subpackage
pip installed snmachine by doing: `pip install -e ./snmachine`
pip install pytest
Unpacked the `SPCC_SUBSET.tar.gz` in the `snmachine/example_data` directory
entered the test directory and ran the tests: `py.test -m "not slow"` and they passed.

Does that work for you?

**Comment by Catarina-Alves** [comment]
> @Catarina-Alves I think this is ready for testing by others. I did the following:
> 
> * Created a new installation of Miniconda with python 3.8
> * Cloned this repo and checked out this branch `general/issue/258/utils-subpackage
> * pip installed snmachine by doing: `pip install -e ./snmachine`
> * pip install pytest
> * Unpacked the `SPCC_SUBSET.tar.gz` in the `snmachine/example_data` directory
> * entered the test directory and ran the tests: `py.test -m "not slow"` and they passed.
> 
> Does that work for you?

Yes, thank you. I will now repeat the tests on my local machine. Once they pass, I will merge the PR and update the pip package.

**Comment by Catarina-Alves** [comment]
I had to update a few libraries in the setup to make all the tests run. I will now merge the PR, and update the version of `snmachine` in pip

**Pull request merged**